### PR TITLE
make X509v3 keyusage optional

### PR DESCRIPTION
**Pull request by sbernard31**
See #748.

### CODE REVIEW

**Comment by boaks** [comment]
Should "C=CA,L=Ottawa,O=Eclipse IoT,OU=Californium,CN=cf-nosigning" really be stored in the "trusts"? 
I don't feel too comfortable to trust it. I would therefore prefer to have it in the keystore.

**Comment by boaks** [comment]
Please mark it, if you want to have it in 2.0.0-M12

File: scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/CertificateRequest.java [hunk 1]
```diff
@@ -444,8 +444,8 @@ boolean isSupportedKeyType(X509Certificate cert) {
 
 		for (ClientCertificateType type : certificateTypes) {
 			boolean isCompatibleType = type.isCompatibleWithKeyAlgorithm(cert.getPublicKey().getAlgorithm());
-			boolean meetsSigningRequirements = !type.requiresSigningCapability() ||
-					(type.requiresSigningCapability() && cert.getKeyUsage() != null && cert.getKeyUsage()[0]);
+			boolean meetsSigningRequirements = !type.requiresSigningCapability()
+					|| (type.requiresSigningCapability() && (cert.getKeyUsage() == null || cert.getKeyUsage()[0]));
```
Comment (boaks) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/CertificateRequest.java]: !a || (a && b) := !a || b

Could this be reduced to the logic equivalent?

Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/CertificateRequest.java]: And please add a comment, that it's appreciated, if someone can provide details about the usage of the "key usage" and "extended key usage" related to a "certificate request".

**Comment by sbernard31** [comment]
> I don't feel too comfortable to trust it. I would therefore prefer to have it in the keystore.

Ok I will do that.

> please add a comment, that it's appreciated, if someone can provide details about the usage of the "key usage" and "extended key usage" related to a "certificate request".

Good idea, I will do that too.

> Could this be reduced to the logic equivalent?

Exact.

**Comment by sbernard31** [comment]
Better now ?

**Comment by boaks** [comment]
LGTM

**Pull request merged**